lib/membrane/rtp/twcc_sender/receiver_rate.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Membrane.RTP.TWCCSender.ReceiverRate do
+  @moduledoc false
+  # module responsible for calculating bitrate
+  # received by a receiver in last `window` time
+  # (referred to as R_hat in the GCC draft, sec. 5.5)
+
+  alias Membrane.Time
+
+  @type t() :: %__MODULE__{
+          value: float() | nil,
+          # time window for measuring the received bitrate, between [0.5, 1]s (reffered to as "T" in the draft)
+          window: Time.t(),
+          # accumulator for packets and their timestamps that have been received in last `window` time
+          packets_received: Qex.t({Time.t(), pos_integer()})
+        }
+
+  @enforce_keys [:window]
+  defstruct @enforce_keys ++ [:value, packets_received: Qex.new()]
+
+  @spec new(Time.t()) :: t()
+  def new(window), do: %__MODULE__{window: window}
+
+  @spec update(t(), Time.t(), [Time.t() | :not_received], [pos_integer()]) :: t()
+  def update(%__MODULE__{value: nil} = rr, reference_time, receive_deltas, packet_sizes) do
+    packets_received = resolve_receive_deltas(receive_deltas, reference_time, packet_sizes)
+
+    packets_received = Qex.join(rr.packets_received, packets_received)
+
+    {first_packet_timestamp, _first_packet_size} = Qex.first!(packets_received)
+    {last_packet_timestamp, _last_packet_size} = Qex.last!(packets_received)
+
+    if last_packet_timestamp - first_packet_timestamp >= rr.window do
+      rr = %__MODULE__{rr | value: 0.0}
+      update(rr, reference_time, receive_deltas, packet_sizes)
```

Comment:
I know it will happen only at the beginning, but we could easily extract everything after line 45 in `update` to eg. `do_update` function that will get already resolved receive deltas and `threshold` as arguments to avoid calculating it twice here.